openedx/features/course_experience/views/course_outline.py

```diff
@@ -30,26 +30,36 @@ class CourseOutlineFragmentView(EdxFragmentView):
     Course outline fragment to be shown in the unified course view.
     """
 
-    def render_to_fragment(self, request, course_id=None, page_context=None, **kwargs):
+    def render_to_fragment(self, request, **kwargs):
         """
         Renders the course outline as a fragment.
         """
+
+        course_id = kwargs['course_id']
         course_key = CourseKey.from_string(course_id)
-        course_overview = get_course_overview_with_access(request.user, 'load', course_key, check_if_enrolled=True)
+        anonymous = request.user.is_anonymous()
+        course_overview = get_course_overview_with_access(
+            request.user, 'load', course_key, check_if_enrolled=not anonymous
+        )
         course = modulestore().get_course(course_key)
 
         course_block_tree = get_course_outline_block_tree(request, course_id)
         if not course_block_tree:
             return None
 
+        enable_links = request.user.is_authenticated or (
```

Comment:
This method may also need a couple of updates for unenrolled but logged-in learners.

Reply:
👍